test/python/fx_importer/all_reduce_test.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2023 Advanced Micro Devices, Inc
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+# RUN: %PYTHON %s | FileCheck %s
+
+import torch
+import torch.multiprocessing as mp 
+import torch.distributed as dist 
+import torch.export
+import torch.nn as nn
+import torch.distributed._functional_collectives as funcol
+import os 
+
+from typing import Optional
+from torch_mlir.extras.fx_importer import FxImporter
+from torch_mlir import ir
+from torch_mlir.dialects import torch as torch_d
+from torch.distributed import init_process_group
+
+def export_and_import(
+    f,
+    *args,
+    fx_importer: Optional[FxImporter] = None,
+    constraints: Optional[torch.export.Constraint] = None,
+    **kwargs,
+):
+    context = ir.Context()
+    torch_d.register_dialect(context)
+
+    if fx_importer is None:
+        fx_importer = FxImporter(context=context)
+    prog = torch.export.export(f, args, kwargs, constraints=constraints)
+    fx_importer.import_frozen_exported_program(prog)
+    return fx_importer.module_op
+
+def test_import_frozen_exported_program():
+    class All_Reduce(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, x):
+            out = funcol.all_reduce(x, "sum", [0, 1, 2, 3])
+            return out
+    x = torch.tensor([1, 2, 3, 4], dtype=torch.float32)
+    m = export_and_import(All_Reduce(), x)
+    print(m)
```

Comment:
You wouldn't have to do so many lit tests. Wrap the print sentence with `if torch.distributed.get_rank() == 0:`, then the IR will only be printed on the first process.